crates/bevy_solari/src/realtime/specular_gi.wgsl

```diff
@@ -51,7 +51,12 @@ fn specular_gi(@builtin(global_invocation_id) global_id: vec3<u32>) {
         wi = wi_tangent.x * T + wi_tangent.y * B + wi_tangent.z * N;
         let pdf = ggx_vndf_pdf(wo_tangent, wi_tangent, surface.material.roughness);
 
-        radiance = trace_glossy_path(surface.world_position, wi, &rng) / pdf;
+        // https://d1qx31qr3h6wln.cloudfront.net/publications/mueller21realtime.pdf#subsection.3.4, equation (4)
+        let cos_theta = saturate(dot(wo, surface.world_normal));
```

Comment:
why is this clamped? paper says its `dot(wo, surface.world_normal)` but I'm not seeing the saturate reason.

Reply:
This is always needed to prevent more than 180 degree angles that can happen when path tracing.